lib/matplotlib/tests/test_figure.py

```diff
@@ -553,6 +553,17 @@ def test_savefig_preserve_layout_engine(tmp_path):
     assert fig.get_layout_engine()._compress
 
 
+def test_savefig_locate_colorbar(tmp_path):
+    fig, ax = plt.subplots()
+    pc = ax.pcolormesh(np.random.randn(2, 2))
+    cbar = fig.colorbar(pc, aspect=40)
+    fig.savefig(tmp_path / 'foo.png', bbox_inches=mpl.transforms.Bbox([[0, 0], [4, 4]]))
```

Comment:
```suggestion
    fig.savefig(io.BytesIO(), bbox_inches=mpl.transforms.Bbox([[0, 0], [4, 4]]))
```

Reply:
Thanks, I didn't know you could do this but I see it is the convention through this module.  So I also took the liberty of updating the test I added at #24971.

Reply:
It is likely faster to stay in memory when saving if possible rather than thrashing the hard drive, which I think is why this would be preferred.